### ISSUE TITLE
release

### ISSUE THREAD

**Issue by ktsaou** [firehol/firehol#107]
@philwhineray @alonbl 

Guys, I want to release all the tools: `firehol`, `iprange`, `netdata`.

Do you believe it is a good time to do it?

**Comment by philwhineray** [comment]
iprange is not a problem. In fact the code has not changed since the 1.0.0 I already did, only the packaging.

I would like to do an extract of all the common version/detection code for `firehol`, `fireqos` et. al. as a final act before doing a 3.0 release. This would be to e.g. `/usr/share/firehol/common.sh`. This way there will be no big changes for packagers soon after. I should be able to do that this weekend if you are OK to wait a few days.

I must confess I have not really looked at netdata much, so I am happy whenever you decide it is ready - if there is something I am needed for (does it need packaging like the others?), let me know.

**Comment by ktsaou** [comment]
ok. nice.

netdata is far from ready.  But it is pretty usable and stable. I have more than a dozen installations of it in production environments without any issues at all. I strongly believe netdata can change our perception of monitoring. Actually, netdata can be a lot more than just system monitoring. I need input though, so... some user input could be nice!

**Comment by alonbl** [comment]
please note that when you release autoconf based package you need to follow the following procedure:
1. set VERSION_SUFFIX to empty ([]).
2. set PACKAGE_RPM_RELEASE to 1
3. add entry to *.spec.in %changelog with current timestamp and release.
4. probably modify change log.
then perform the commit of release ("release 1.0.0") and tag xxx-1.0.0
immediately after prepare master for next release:
0. increment VERSION_FIX
1. revert VERSION_SUFFIX to _master
2. revert PACKAGE_RPM_RELEASE to 0.0.$(echo VERSION_SUFFIX | sed 's/^_//')
commit ("post release 1.0.0").

**Comment by philwhineray** [comment]
OK I pushed my changes to split out the common functions in firehol and the others.

There is a bit more tidy-up to do (e.g. move bash 4 check to configure script) and I will try to bring all of the release systems in line with Alon's proposal above as part of the release.

**Comment by philwhineray** [comment]
I have been working on making the firehol release closer to the list above.

Should we be pushing out rpm-specific files? In general I think it would be better to leave packaging matters to the distributions and stick to tar-files.

In particular I just eliminated NEWS, so now there is only ChangeLog with version/release info in them. I don't really want to have to amend a file I cannot check.

**Comment by alonbl** [comment]
> Should we be pushing out rpm-specific files? In general I think it would be better to leave packaging matters to the distributions and stick to tar-files.

Correct, you should only distribute the "make dist" tarball out of the release tag. Downstream may use the spec file as a reference, people can rpmbuild -tb TARBALL for their convenience.

**Comment by ktsaou** [comment]
Keep in mind I fixed a bug in `iprange`. We should release the latest version.

Also, give me this weekend to make a few changes to `netdata`. I am working on it.

**Comment by philwhineray** [comment]
OK I pushed out an `iprange 1.0.1`.

Most of the steps that @alonbl suggested are now automated in the hooks. Just remove the suffix from the `configure.ac` and make the `ChangeLog` and `spec.in` updates and commit.

Provided everything checks and a test build succeeds then the commit is allowed and a signed tag initiated automatically, followed by resetting the versions in configure.ac and committing automatically. Nothing is pushed so it is always possible to revert locally if there is a problem.

Requires you to have a gpg keys for signing and there are still steps needed to complete the release on the server - I will try to document these more thoroughly and if you still want a webinar I will go through it all.

FireHOL 3.0.0 will follow either tonight or tomorrow.

**Comment by alonbl** [comment]
Great!

You should attach the make dist tarball into the release at[1]

[1] https://github.com/firehol/iprange/releases

**Comment by philwhineray** [comment]
Unless it is easy to automate I am comfortable with the current location:

http://firehol.org/download/iprange/releases/

I probably need to fix the website to account for multiple projects before
long.

**Comment by alonbl** [comment]
ok, although github outputs are expected to be at github... maybe add a text to release referencing the webdav?

also, if you are going to distribute xz, you do not need the bz2, please also update the Source: at spec.in file to use the xz.

please also update the Source: at spec.in to be full url to where you decide to publish the tarball.

Thanks!

**Comment by philwhineray** [comment]
Looks like it can be automated with a bit of effort.

For now I have added a reference but I think it should be possible to copy
the results over programatically at some point.

I will update the spec.in when I have had time to sort it out or not.

**Comment by philwhineray** [comment]
With netdata the following exist in git but do not make it into the tar file:

* `conf.d/charts.d.conf`
* `src/old-and-tests/netdata.c`,  `src/old-and-tests/benchmark.c`,  `src/old-and-tests/benchmark2.c`

When I come to tag a release should I ignore these or are they files which should be either installed or deleted from the repo?

**Comment by ktsaou** [comment]
@philwhineray this is ok. Ignore them. I just kept them there.

I'll open an issue in netdata to update you for its latest development.
